utils/config_build_and_install.macos.sh

```diff
@@ -6,7 +6,7 @@ then
 fi
 
 export PIP_DEPENDENCIES="Cython coverage nose numpy matplotlib"
-if $WITH_ROS_SUPPORT && [ -z $ROS_DISTRO ]
```

Comment:
I think this one is an attempt at supporting "source ROS" on macOS and should be left as `$ROS_DISTRO`